lib/rubocop/cop/rspec/pending.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks for any pending or skipped examples.
+      #
+      # @example
+      #   # bad
+      #   describe MyClass do
+      #     it "should be true"
+      #   end
+      #
+      #   describe MyClass do
+      #     it "should be true" do
+      #       pending
+      #     end
+      #   end
+      #
+      #   describe MyClass do
+      #     xit "should be true" do
+      #     end
+      #   end
+      #
+      #   # good
+      #   describe MyClass do
+      #   end
+      class Pending < Cop
+        MSG = 'Pending spec found.'.freeze
+
+        skippable = ExampleGroups::GROUPS + Examples::EXAMPLES
+
+        SKIPPABLE_SELECTORS = skippable.node_pattern_union
+
+        SKIP_SYMBOL = s(:sym, :skip)
+        SKIP_TRUE   = s(:pair, SKIP_SYMBOL, s(:true))
+
+        def_node_matcher :metadata, <<-PATTERN
+          {(send nil? #{SKIPPABLE_SELECTORS} ... (hash $...))
+           (send nil? #{SKIPPABLE_SELECTORS} $...)}
+        PATTERN
```

Comment:
I’m having a hard time reading this `:metadata` node matcher, together with the `grep` methods in the last method. Would it be possible to make it more explicit how things work?

I believe all of these metadata examples should be matched:

- `:skip`
- `skip: true`
- `skip: "does not work on CI"`
- `:pending`
- `pending: true`
- `pending: "fix it later"`

Even dynamic values should work, e.g. `skip: !ENV['CI']`, right?

Reply:
I refactored the code a bit to help with the `grep`s, but in regard to that list of examples, I think only the second and third are supported.  I can't find any rspec docs that say you can use `pending` as metadata, just `skip`.  If you find something that states otherwise let me know and I'll update accordingly. 

As far as the last point, I'm not sure... I can't think of how a static code analyzer could ever deal with that in an expected manner

Reply:
> I can't find any rspec docs that say you can use `pending` as metadata, just `skip`. If you find something that states otherwise let me know and I'll update accordingly.

https://relishapp.com/rspec/rspec-core/v/3-7/docs/pending-and-skipped-examples/pending-examples#`pending`-for-an-example-that-is-currently-passing-with-a-reason shows the example `example("something else getting finished", :pending => 'unimplemented') do`…

I actually can’t find any documentation for the `:skip` and `:pending` (symbol) metadata, only the hash variants.

Reply:
According [to the docs](https://relishapp.com/rspec/rspec-core/v/3-7/docs/pending-and-skipped-examples), the `skip` behavior is to prevent a spec from running for whatever reason, whereas `pending` is for specs that are _known_ to fail and are still run which I think are really distinct use cases especially with regard to a cop. Could we consider introducing separate cops for each use case?

Reply:
That's a good point @phallguy - I could see the argument for splitting them up, though I'll admit I don't feel very strongly either way.

Reply:
Yeah I could be convinced either way. There are a few real world scenarios I'm thinking about:

1. `skip` because something broke and I don't want to fix it.
1. `skip` because I just wrote an  spec while debugging to make sure I cover a new edge case I discovered but forgot to go back and actually write the spec.
1. `pending` because we're refactoring and this is expected to fail in the interim but it still represents logic that we want covered and if it suddenly starts working when we weren't expecting it to thats a problem.

I'd like to be able to block checkin or at least alert in CI based on 1 & 2 but allow 3 with some sort of TODO comment with ticket/kanban link.

I suppose I could achieve what I want with a `#robcop disable ....` comment on the `pending` specs so definitely not a blocker.

Also wanted to confirm if 'empty' specs will be caught as well. I believe the following is also considered a pending spec (notice no block):

```ruby
it "checks a new condition I just thought of"
```

and prefixing with an _x_ should also be considered pending.
```ruby
xit "used to work but whatever YOLO!" do
  #...
end
```

Reply:
Hmm I'm not sure if it will catch empty blocks and report them.  I can look into adding that functionality, but if there are any other features I'd prefer to explore planning them for future work and not part of this PR?  It's starting to become difficult to continue pouring time into this

Reply:
Yeah no worries. Thank you for the work you've already done!